test/test_object_storage_azure.py

```diff
@@ -0,0 +1,77 @@
+"""Copyright (c) 2022 Aiven, Helsinki, Finland. https://aiven.io/"""
+from io import BytesIO
+from tempfile import NamedTemporaryFile
+from unittest.mock import MagicMock
+
+import pytest
+import sys
+
+
+@pytest.fixture(scope="module", name="mock_azure_module")
+def fixture_mock_azure_module():
+    get_blob_client_mock = MagicMock()
+    blob_client = MagicMock(get_blob_client=get_blob_client_mock)
+    service_client = MagicMock(from_connection_string=MagicMock(return_value=blob_client))
+    sys.modules["azure.common"] = MagicMock()
+    sys.modules["azure.core.exceptions"] = MagicMock()
+    sys.modules["azure.storage.blob"] = MagicMock(BlobServiceClient=service_client)
```

Comment:
This manually alters global variables in a remote module and does not revert it at the end of the test.
This should use something like [mock.patch](https://docs.python.org/dev/library/unittest.mock.html#unittest.mock.patch).

Reply:
`patch.dict()` seems like the simplest way to do the trick.